scripts/watcher.js

```diff
@@ -0,0 +1,28 @@
+const chokidar  = require('chokidar')
+const { exec } = require('child_process')
+const chalk = require('chalk')
+const log = console.log
+
+const watcher = chokidar.watch('packages/**/src/*.ts', { 
+  persistent: true ,
+  usePolling: true,
+  interval: 500
+})
+
+log(chalk.yellow.bold('Watching all files... 👀'))
+
+watcher.on('change', filePath => {
+  log(chalk.redBright.bold(`${filePath} is changed`))
+
+  const path = filePath.split('/')
+  const location = `${path[0]}/${path[1]}/`
+
+  exec(`tsc`, { cwd: location}, (err, stdout) => {
+    if (!err ||  err === null) {
+      log(chalk.greenBright(`Package ${path[1]} is re-built`))
+    } else {
+      log(chalk.white(err))
```

Comment:
@alexnm I added these lines to throw errors when compilation fails, I tested this our for some time and its smooth enough.

Reply:
cool, eager to test this out also :)

Reply:
I am just pushing one more commit, by adding ``incremental`` in **tsconfig.json** that will decrease the compilation time. Please try with that